### PR TITLE
Add specs for OpenSSL::KDF.scrypt

### PR DESCRIPTION
**Pull request by herwinw**

### CODE REVIEW

File: library/openssl/kdf/scrypt_spec.rb [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    key = OpenSSL::KDF.scrypt("", **@defaults)
+    key.should == "\xAA\xFC\xF5^E\x94v\xFFk\xE6\xF0vR\xE7\x13\xA7\xF5\x15'\x9A\xE4C\x9Dn\x18F_E\xD2\v\e\xB3".b
+  end
+
+  it "accepts an empty salt" do
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, salt: "")
+    key.should == "\x96\xACDl\xCB3/aN\xB0F\x8A#\xD7\x92\xD2O\x1E\v\xBB\xCE\xC0\xAA\xB9\x0F]\xB09\xEA8\xDD\e".b
+  end
+
+  it "accepts an empty length" do
```
Comment (andrykonchin) [inline library/openssl/kdf/scrypt_spec.rb:77]:
```suggestion
  it "accepts a zero length" do
```

**Comment by andrykonchin** [comment]
Looks good to me!

File: library/openssl/kdf/scrypt_spec.rb (hunk 1 above)
Comment (herwinw) [reply library/openssl/kdf/scrypt_spec.rb]: Updated. That does sound like the better description. (I probably copy-pasted the description above, where "empty" makes sense)

**Comment by andrykonchin** [comment]
Thank you!

**Comment by eregon** [comment]
@herwinw I'm seeing errors like this in TruffleRuby's CI
```
1)
OpenSSL::KDF.scrypt creates the same value with the same input ERROR
NoMethodError: undefined method `scrypt' for OpenSSL::KDF:Module

    key = OpenSSL::KDF.scrypt("secret", **@defaults)
                      ^^^^^^^
/b/b/e/main/spec/ruby/library/openssl/kdf/scrypt_spec.rb:16:in `block (2 levels) in <top (required)>'
/b/b/e/main/spec/ruby/library/openssl/kdf/scrypt_spec.rb:4:in `<top (required)>'

2)
OpenSSL::KDF.scrypt supports nullbytes embedded into the password ERROR
NoMethodError: undefined method `scrypt' for OpenSSL::KDF:Module

    key = OpenSSL::KDF.scrypt("sec\x00ret".b, **@defaults)
                      ^^^^^^^
/b/b/e/main/spec/ruby/library/openssl/kdf/scrypt_spec.rb:21:in `block (2 levels) in <top (required)>'
/b/b/e/main/spec/ruby/library/openssl/kdf/scrypt_spec.rb:4:in `<top (required)>'
```
on `ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]` (so on CRuby)
on Oracle Linux 7 which has `OpenSSL 1.0.2k-fips  26 Jan 2017`.
Is `scrypt` something from OpenSSL 1.1 or so?

**Comment by rhenium** [comment]
> on Oracle Linux 7 which has `OpenSSL 1.0.2k-fips 26 Jan 2017`.
> Is `scrypt` something from OpenSSL 1.1 or so?

Yes - scrypt is OpenSSL >= 1.1.0. I think it's not implemented in current versiosn of LibreSSL either.

**Pull request merged**